HMSegmentedControl/HMSegmentedControl.m

```diff
@@ -756,31 +776,63 @@ - (NSUInteger)sectionCount {
                self.type == HMSegmentedControlTypeTextImages) {
         return self.sectionImages.count;
     }
-    
+
     return 0;
 }
 
 - (void)willMoveToSuperview:(UIView *)newSuperview {
     // Control is being removed
     if (newSuperview == nil)
         return;
-    
+
     if (self.sectionTitles || self.sectionImages) {
         [self updateSegmentsRects];
     }
 }
 
 #pragma mark - Touch
 
+- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event {
+    UITouch *touch = [touches anyObject];
+    // Get the index of the selected segment
+    NSInteger segment = [self calculateSelectedSegmentPosition:touch];
+    // Show pressed down state
+    [self.touchDownIndicatorLayers[segment] setOpacity: 1.0];
+}
+
 - (void)touchesEnded:(NSSet *)touches withEvent:(UIEvent *)event {
     UITouch *touch = [touches anyObject];
+    // Get the index of the selected segment
+    NSInteger segment = [self calculateSelectedSegmentPosition:touch];
```

Comment:
If you press down, slide your finger off the segment, and then up, I think the segment will remain selected. You might need to support touch moving (or maybe when touches end, you can just remove all the touch down indicators).

Reply:
I _think_ this gets handled by HMScrollView. There's an implementation of touchesBegan/touchesEnded at the top of this file that checks if there's dragging and then forwards the touch to the responder.